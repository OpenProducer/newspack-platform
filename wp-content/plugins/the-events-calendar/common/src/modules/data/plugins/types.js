/**
 * Internal dependencies
 */
import { PREFIX_COMMON_STORE } from '@moderntribe/common/data/utils';

export const ADD_PLUGIN = `${ PREFIX_COMMON_STORE }/ADD_PLUGIN`;
export const REMOVE_PLUGIN = `${ PREFIX_COMMON_STORE }/REMOVE_PLUGIN`;
