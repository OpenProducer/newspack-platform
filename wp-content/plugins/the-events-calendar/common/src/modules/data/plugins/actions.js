/**
 * Internal dependencies
 */
import * as types from './types';

export const addPlugin = ( name ) => ( {
	type: types.ADD_PLUGIN,
	payload: {
		name,
	},
} );

export const removePlugin = ( name ) => ( {
	type: types.REMOVE_PLUGIN,
	payload: {
		name,
	},
} );
