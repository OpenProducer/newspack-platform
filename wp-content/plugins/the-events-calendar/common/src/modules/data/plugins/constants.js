export const EVENTS_PLUGIN = 'events';
export const EVENTS_PRO_PLUGIN = 'events-pro';
export const TICKETS = 'tickets';
export const TICKETS_PLUS = 'tickets-plus';