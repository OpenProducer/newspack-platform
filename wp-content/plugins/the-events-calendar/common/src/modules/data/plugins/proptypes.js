/**
 * External Dependencies
 */
import PropTypes from 'prop-types';

//
// ─── VENDOR ─────────────────────────────────────────────────────────────────────
//

export const ReactSelectOption = PropTypes.shape( {
	label: PropTypes.string.isRequired,
	value: PropTypes.any.isRequired,
} );

export const ReactSelectOptions = PropTypes.arrayOf( ReactSelectOption );
