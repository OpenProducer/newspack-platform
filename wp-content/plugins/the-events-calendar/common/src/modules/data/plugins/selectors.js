/**
 * External dependencies
 */
import { includes } from 'lodash';
import { curry } from 'lodash/fp';

export const getPlugins = ( state ) => state.plugins;

export const hasPlugin = curry( ( state, plugin ) => includes( getPlugins( state ), plugin ) );
