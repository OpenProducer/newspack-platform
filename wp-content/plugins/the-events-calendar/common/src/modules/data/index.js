/**
 * Internal Dependencies
 */
import * as editor from './editor/post-types';
import * as forms from './forms';
import * as plugins from './plugins';
export { default } from './reducers';

export { editor, forms, plugins };