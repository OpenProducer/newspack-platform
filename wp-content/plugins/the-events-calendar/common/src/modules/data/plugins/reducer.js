/**
 * External dependencies
 */
import { uniq } from 'lodash';

/**
 * Internal dependencies
 */
import { types } from '@moderntribe/common/data/plugins';

export default ( state = [], action ) => {
	switch ( action.type ) {
		case types.ADD_PLUGIN:
			return uniq( [ ...state, action.payload.name ] );
		case types.REMOVE_PLUGIN:
			return [ ...state ].filter( ( pluginName ) => pluginName !== action.payload.name );
		default:
			return state;
	}
};
