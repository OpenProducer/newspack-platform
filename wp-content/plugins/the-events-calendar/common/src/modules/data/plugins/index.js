/**
 * Internal dependencies
 */
import reducer from './reducer';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import * as constants from './constants';
import * as proptypes from './proptypes';

export default reducer;
export { types, actions, selectors, constants, proptypes };
