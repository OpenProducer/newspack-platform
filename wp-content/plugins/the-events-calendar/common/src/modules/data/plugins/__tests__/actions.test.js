/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/common/data/plugins';

describe( 'Plugin actions', () => {
	test( 'Add Plugin', () => {
		expect( actions.addPlugin( 'events' ) ).toMatchSnapshot();
	} );

	test( 'Remove Plugin', () => {
		expect( actions.removePlugin( 'events' ) ).toMatchSnapshot();
	} );
} );
