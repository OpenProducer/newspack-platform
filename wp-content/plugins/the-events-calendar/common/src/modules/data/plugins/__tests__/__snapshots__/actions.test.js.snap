// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plugin actions Add Plugin 1`] = `
Object {
  "payload": Object {
    "name": "events",
  },
  "type": "@@MT/COMMON/ADD_PLUGIN",
}
`;

exports[`Plugin actions Remove Plugin 1`] = `
Object {
  "payload": Object {
    "name": "events",
  },
  "type": "@@MT/COMMON/REMOVE_PLUGIN",
}
`;
